Modules/_abc.c

```diff
@@ -446,6 +449,27 @@ _abc__abc_init(PyObject *module, PyObject *self)
         return NULL;
     }
     Py_DECREF(data);
+    if (PyType_Check(self)) {
+        PyTypeObject *cls = (PyTypeObject *)self;
+        PyObject *flags = _PyDict_GetItemIdWithError(cls->tp_dict, &PyId___flags__);
```

Comment:
Hm, this whole mechanism makes me a bit uncomfortable. I'm not aware of any case where writing to `__flags__` in Pythonland actually works like this.

But I can't really think of anything better, so I guess it's fine. Maybe we could just use a name like `_abc_tpflags` or something instead of `__flags__`?

Reply:
Sure, it is just a name. I like using a dunder name, as it is a special thing. `__abc_tpflags__` perhaps?